sdks/java/io/redis/src/main/java/org/apache/beam/sdk/io/redis/RedisIO.java

```diff
@@ -740,22 +739,21 @@ private void writeUsingDecrBy(KV<String, String> record, Long expireTime) {
         String key = record.getKey();
         String value = record.getValue();
         long decr = Long.parseLong(value);
-        pipeline.decrBy(key, decr);
+        transaction.decrBy(key, decr);
 
         setExpireTimeWhenRequired(key, expireTime);
       }
 
       private void setExpireTimeWhenRequired(String key, Long expireTime) {
         if (expireTime != null) {
-          pipeline.pexpire(key, expireTime);
+          transaction.pexpire(key, expireTime);
         }
       }
 
       @FinishBundle
       public void finishBundle() {
-        if (pipeline.isInMulti()) {
-          pipeline.exec();
-          pipeline.sync();
+        if (transaction != null) {
```

Comment:
How about checking `batchCount` here if `exec()` is needed? And afterwards transaction could be set to `null`

Reply:
I updated!